paimon-core/src/main/java/org/apache/paimon/catalog/AbstractCatalog.java

```diff
@@ -205,4 +209,11 @@ public static Path databasePath(String warehouse, String database) {
     protected boolean isSystemDatabase(String database) {
         return SYSTEM_DATABASE_NAME.equals(database);
     }
+
+    protected boolean allowCreateTableInDefaultDB(Identifier identifier) {
+        return this.options
+                .getOptional(DISABLE_CREATE_TABLE_IN_DEFAULT_DB)
+                .map(val -> !(val & DEFAULT_DATABASE.equals(identifier.getDatabaseName())))
```

Comment:
Use `&&`？

Reply:
In addition, users can configure default database name with option `default-database` for flink